Tests/Cosmos.Compiler.Tests.Bcl/System/Collections/Not Generic/HashtableTest.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections;
+using Cosmos.TestRunner;
+//using Cosmos.Compiler.Tests.Bcl.Helper;
+using Cosmos.Debug.Kernel;
+
+namespace Cosmos.Compiler.Tests.Bcl.System.Collections.Not_Generic
+{
+    class HashtableTest
+    {
+        private static Debugger myDebugger = new Debugger("System", "HashtableTest");
+
+        public static void Execute()
+        {
+            var h = new Hashtable();
+            Assert.IsTrue(h != null, "Hashtable ctor returns but h is null");
+
+            h.Add("One", "One");
+            //h.Add(42, "Test");
+
+            Assert.IsTrue(h.ContainsKey("One"), "Hashtable.ContainsKey() failed: existing key not found");
+            Assert.IsFalse(h.ContainsKey("Two"), "Hashtable.ContainsKey() failed: not existing key not found");
+
+            Assert.IsTrue((string)h["One"] == "One", "Hashtable indexer failed: existing value not found");
+            Assert.IsTrue(h["Two"] == null, "Hashtable indexer failed: not existing value not found");
+
+            /* The indexer written in this way should be the same thing of Add("Two", "Two") */
+            h["Two"] = "Two";
+
+            Assert.IsTrue((string)h["Two"] == "Two", "Hashtable indexer failed: existing value (II) not found");
+
+            Assert.IsTrue(h.Count == 2, "Hashtable Count failed: value != 2");
+
+            /*
+             * Got Il2CPU exception:
+             * System.Exception: Original method argument $this is a reference type. Plug attribute first argument is not an argument type, nor was it marked with ObjectPointerAccessAttribute! Method: SystemObjectSystemArrayGetValueSystemInt32 Parameter: aThis
```

Comment:
The fix for this is simple. Change [this line](https://github.com/CosmosOS/Cosmos/blob/master/source/Cosmos.Core_Plugs/System/ArrayImpl.cs#L95) to:
```cs
 public static unsafe uint GetValue([ObjectPointerAccess]uint* aThis, int aIndex)
```
It's probably a good idea to apply that change to all methods in `ArrayImpl`.